### PR TITLE
Fix trtl store tests race condition

### PR DESCRIPTION
**Pull request by pdeziel**
### Scope of changes

We have been seeing a race condition in the CI in the trtl store tests. After some debugging it seems we were misplacing the pointer to the trtl server and were therefore unable to close it during test suite shutdown.

This also fixes some server-side errors appearing in the tests that were not the cause of the race condition but were still logging errors in the tests, due to iterators that were not fully consumed.

SC-7652

### Type of change

- [x] bug fix
- [ ] new feature
- [ ] documentation
- [ ] other (describe)

### Acceptance criteria

Hopefully this gets rid of that race condition once and for all!

### Author checklist

- [x] I have manually tested the change and/or added automation in the form of unit tests or integration tests
- [ ]  I have updated the dependencies list
- [ ]  I have recompiled and included new protocol buffers to reflect changes I made
- [ ]  I have added new test fixtures as needed to support added tests
- [x]   Check this box if a reviewer can merge this pull request after approval (leave it unchecked if you want to do it yourself)
- [x]  I have moved the associated Shortcut story to "Ready for Review"

### Reviewer(s) checklist

- [ ] Any new user-facing content that has been added for this PR has been QA'ed to ensure correct grammar, spelling, and understandability.

### CODE REVIEW

File: pkg/gds/store/trtl/trtl_test.go [hunk 1]
```diff
@@ -76,11 +76,11 @@ func (s *trtlStoreTestSuite) SetupSuite() {
 	require.NoError(err)
 	s.conf = &conf
 
-	trtl, err := trtl.New(*s.conf)
+	s.trtl, err = trtl.New(*s.conf)
```
Comment (pdeziel) [inline pkg/gds/store/trtl/trtl_test.go:79]: 🤦‍♂️

Comment (rebeccabilbro) [reply pkg/gds/store/trtl/trtl_test.go:79]: hehe oops

File: pkg/gds/store/trtl/trtl_test.go [hunk 2]
```diff
@@ -76,11 +76,11 @@ func (s *trtlStoreTestSuite) SetupSuite() {
...
 	require.NoError(err)
 
 	s.grpc = bufconn.New(bufSize, "")
-	go trtl.Run(s.grpc.Listener)
+	go s.trtl.Run(s.grpc.Listener)
```
Comment (rebeccabilbro) [inline pkg/gds/store/trtl/trtl_test.go:83]: 💪

**Review by rebeccabilbro — APPROVED** [review]
Amazing catch!

**Pull request merged**